### PR TITLE
Add ssTEM dataset

### PR DESCRIPTION
**Pull request by adamjstewart**
This PR adds a dataset for [ISBI Challenge: Segmentation of neuronal structures in EM stacks](http://brainiac2.mit.edu/isbi_challenge/), which involves imagery from an ssTEM microscope.

Questions:

- I wasn't sure what to call the dataset, as there are several different ISBI challenges per year
- ~This dataset requires `skimage` to load, as it includes multi-channel TIF images that PIL does not support. How should we handle this? Should I make the `skimage` import lazy so that it isn't required to import `torchvision`?~
- ~This is a segmentation dataset. As such, most transforms like RandomHorizontalFlip and RandomAffine must be applied to both img and target. Is their any precedence for how to handle this? Or should I have a separate `transform` and `target_transform` like other datasets?~
- The train dataset comes with both data and labels, but the test dataset only comes with data. Users are expected to upload their predictions to the challenge website and have them graded for them. Hopefully this is okay.

### CODE REVIEW

File: torchvision/datasets/sstem.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+from __future__ import print_function
+
+import os
+
+from PIL import Image
+import skimage
```
Comment (pmeier) [inline torchvision/datasets/sstem.py]: This would make `torchvision` dependent on `skimage` at import. Better do a lazy `import` as you suggested.

File: torchvision/datasets/sstem.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import torch.utils.data as data
+
+from .utils import download_url, check_integrity
+
+
+class ssTEM(data.Dataset):
```
Comment (pmeier) [inline torchvision/datasets/sstem.py]: You should subclass the `VisionDataset` from `torchvision.datasets.vision`.

File: torchvision/datasets/sstem.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            }
+        }
+    }
+
+    def __init__(self, root, train=True, transform=None, target_transform=None,
+                 download=False):
+        self.root = os.path.expanduser(root)
+        self.train = train
+        self.transform = transform
+        self.target_transform = target_transform
```
Comment (pmeier) [inline torchvision/datasets/sstem.py]: The separate handling of `transform` and `target_transform` is only for backward compatibility. You should use `transforms` now. See #872.

**Comment by pmeier** [comment]
> - The train dataset comes with both data and labels, but the test dataset only comes with data. Users are expected to upload their predictions to the challenge website and have them graded for them. Hopefully this is okay.

I'm not sure about this. IMO we should not have a dataset (or in this case a split of the dataset) without labels. @fmassa ?

File: torchvision/datasets/sstem.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                'md5': '9767660d7abe4e0ecbdd0061a16058ad'
+            }
+        }
+    }
+
+    def __init__(self, root, train=True, transforms=None, download=False):
+        # Lazy import
+        import skimage.io
+
+        self.root = os.path.expanduser(root)
```
Comment (pmeier) [inline torchvision/datasets/sstem.py]: Please call `__init__` of `VisionDataset` within  the constructor

File: torchvision/datasets/sstem.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        }
+    }
+
+    def __init__(self, root, train=True, transforms=None, download=False):
+        # Lazy import
+        import skimage.io
+
+        super(ssTEM, self).__init__(root, transforms)
+
+        self.root = os.path.expanduser(root)
```
Comment (pmeier) [inline torchvision/datasets/sstem.py]: With the [call of the superclass constructor](https://github.com/pytorch/vision/blob/50d54a82d1479ffb6dd7469ed05fccdf290a1d84/torchvision/datasets/vision.py#L10-L12), this is now obsolete.

File: torchvision/datasets/sstem.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        self.train = train
+        self.transforms = transforms
```
Comment (pmeier) [inline torchvision/datasets/sstem.py]: See comment above.

**Comment by adamjstewart** [comment]
@fmassa What's the status of this PR? Is this something that is still wanted, or should I close it?

**Comment by adamjstewart** [comment]
It seems like this PR has been abandoned by upstream, so I'm going to close it. Feel free to reopen or steal these commits to make a new PR.
